azure_role_definition scope case insensitivity not caught in plan

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.5.2

### AzureRM Provider Version

3.73.0

### Affected Resource(s)/Data Source(s)

azurerm_role_definition

### Terraform Configuration Files

```hcl
resource "azurerm_role_definition" "custom_role" {
  name        = "custom_role"

  scope = "/providers/Microsoft.Management/managementgroups/<mg-id>>"
  permissions {
    actions = [
      "Microsoft.Authorization/*/Read"
    ]
    not_actions = []
  }
}
```


### Debug Output/Panic Output

```shell
azurerm_role_definition.packet_capture: Creating...
╷
│ Error: failed to parse scope from Role Definition ID "/providers/Microsoft.Authorization/roleDefinitions/<rd-id>|/providers/Microsoft.Management/managementgroups/<mg-id>"
│ 
│   with azurerm_role_definition.packet_capture,
│   on roles.tf line 1, in resource "azurerm_role_definition" "custom_role":
│    1: resource "azurerm_role_definition" "custom_role" {
│ 
╵
Error: Terraform exited with code 1.
Error: Process completed with exit code 1.
```


### Expected Behaviour

Given a lot of azurerm resources are either case insensitive or raise when the scope is invalid, the terraform plan should have either caught that  `managementgroups` (rather than `managementGroups`) was not a valid scope segment or handled the casing.

### Actual Behaviour

The plan was successful and the custom role was created in azure on terraform apply, however the terraform apply failed on trying to read the resource back in. This meant that subsequent plans were also broken and the custom role definition had to be manually removed. 

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/authorization/role_definition_resource.go
+++ b/internal/services/authorization/role_definition_resource.go
@@ -17,6 +17,7 @@ import (
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/authorization/migration"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/authorization/parse"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
 	"github.com/hashicorp/terraform-provider-azurerm/utils"
 )
@@ -59,9 +60,10 @@ func resourceArmRoleDefinition() *pluginsdk.Resource {
 			},
 
 			"scope": {
-				Type:     pluginsdk.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringStartsWithOneOf("/subscriptions/", "/providers/Microsoft.Management/managementGroups/"),
 			},
 
 			"description": {

--- a/internal/tf/validation/pluginsdk.go
+++ b/internal/tf/validation/pluginsdk.go
@@ -292,3 +292,20 @@ func StringMatch(r *regexp.Regexp, message string) func(interface{}, string) ([]
 func StringNotInSlice(invalid []string, ignoreCase bool) func(interface{}, string) ([]string, []error) {
 	return validation.StringNotInSlice(invalid, ignoreCase)
 }
+
+func StringStartsWithOneOf(prefixs ...string) func(interface{}, string) ([]string, []error) {
+	return func(i interface{}, k string) (warnings []string, errors []error) {
+		v, ok := i.(string)
+		if !ok {
+			errors = append(errors, fmt.Errorf("expected type of %s to be string", k))
+			return warnings, errors
+		}
+		for _, val := range prefixs {
+			if strings.HasPrefix(v, val) {
+				return warnings, errors
+			}
+		}
+		errors = append(errors, fmt.Errorf("expect %s to start with one of %s, got %q", k, strings.Join(prefixs, ", "), v))
+		return warnings, errors
+	}
+}

